src/control/destination/endpoint.rs

```diff
@@ -36,6 +36,13 @@ impl Endpoint {
         self.metadata.dst_labels()
     }
 
+    pub fn can_speak_h2(&self) -> bool {
```

Comment:
This might be a slightly misleading name --- the proxy can communicate with h2 endpoints that are not other proxies (grpc services for instance).  Is this `prefers_h2()` clearer?

Reply:
I've been finding it confusing to just say that it prefers or can use h2, when really, we want to know if it can use h2 *and* translate with our orig-proto headers. So, I've changed this method to `can_use_orig_proto`.